### PR TITLE
User Guide - setting master as stable

### PR DESCRIPTION
**Pull request by seanmadsen**
Ref: https://github.com/civicrm/civicrm-user-guide/issues/146

### CODE REVIEW

**Comment by nganivet** [comment]
The way we use the 'stable' label today is very confusing and absolutely wrong.

Since all developments happen on the 'master' branch, it certainly cannot be considered 'stable'. By definition 'stable' means that the the software's behavior, functionality, specification or API is considered ‘final’ for that version.  Apart from security patches and bug fixes, the software will not change (cf. https://bitdepth.thomasrutter.com/2010/04/02/stable-vs-stable-what-stable-means-in-software/ and many other references on the Internet).

So, for CiviCRM, we should have the following labels for our branches:
- 'latest': 4.7, master
- 'stable': 4.6
- 'historical': 4.6, 4.7

**Comment by nganivet** [comment]
@totten @colemanw please see above comment. It always bugged me that the page https://civicrm.org/download/list makes the same mistake. Out of pure respect for people that are downloading the software, we should not label 4.7 as 'stable' and then bring-in new features in each monthly release. People that download the 'stable' release expect something else. The LExIM strategy does not change this in any way since we still have the 'IM' part of it.

So let's make this cleat in the documentation and on our download page. Thanks.

**Comment by seanmadsen** [comment]
@nganivet I think there are two separate questions here: 

1. Which release does the CiviCRM project currently call "stable"? Is it `4.6`? Or `4.7`?
2. Should the User Guide offer separate editions with *different content* for a `4.7` book vs `master` book?

I don't have an opinion about the first question, but I do think that the docs should follow the language used in the releases of the project. As you point out, the releases say `4.7` is "stable", so I think it's appropriate for the documentation to do so as well. If other people decide to change this language and call `4.6` "stable", then I'm happy to follow along and change the docs to say `4.6` is "stable". Basically I think this question is not a docs question but a project question, and personally I want to stay out of that discussion so I can focus my time elsewhere. 

The second question is what I'm after here. My opinion is "no", and I feel rather strongly about it. As I see it, we've got a pretty major problem with the user guide which is that for the past 10 months [all of the edits two it](https://github.com/civicrm/civicrm-user-guide/compare/4.7...master) have been going into a version which is actually *not discoverable anywhere*. I [brought this issue up](https://github.com/civicrm/civicrm-user-guide/issues/146) almost a month ago but have heard very little feedback from anyone about it. The change in this PR that I just merged (which is actually not live yet, until someone runs `git pull` for `civicrm-docs` on the server) implements the specific changes which Michael [approved](https://github.com/civicrm/civicrm-user-guide/issues/146#issuecomment-283499434) 11 days ago. 

@nganivet, from you comments above, I understand your position on question 1. But I'm curious to hear your opinion on question 2. Do you think we need separate book content for `4.7` and `master`? If so, what would the difference be between these books? If not, then we agree, and the only sticking point here is question 1, which I'm really not sure how to address here, because I think it's outside the decision-making scope of this issue tracker.

It's also worth noting that in #50 I'm proposing some changes which add quite a bit more flexibility to the way we describe different versions of documentation. I actually envision a setup where we don't even use the term "stable" (because as the link you provided mentions, different people have different interpretations of this word). 

Most importantly, I'm trying get the latest edits into the version of the book which people are reading (at the URL with "stable" in the path, as publicized [here](https://civicrm.org/documentation). If you are suggesting specific changes to the way we set up the books, then I am curious to hear your proposed solution for addressing this problem that we currently face.

**Comment by nganivet** [comment]
I fully appreciate that you do not want to get involved in 1), hence my mention to Tim/Coleman right after commenting on your PR. I will sort this out with them.

Regarding 2), IMO we should have the following public paths for the documentation (looking at only the last snippet):
- 4.7 and latest should both point to the documentation related to the master branch
- I do not care if master exists, but if it does it should point to the master branch as well
- 4.6 and should point to the documentation related to the 4.6 branch
- I do not care if stable exists, but if it does it should point to the LTS branch (ie. 4.6 currently)

So we definitely need to change the page https://civicrm.org/documentation as well.

**Comment by nganivet** [comment]
(reading your comment on #50, 'latest' in the above could be replaced by 'current' just as well)

**Comment by seanmadsen** [comment]
Ok, @nganivet, I'll plan to leave the docs as-is though (with `/user/en/stable` still pointing to the docs for CiviCRM 4.7, as it has been since before the commits in this PR), and if we get the word from the core team that the terminology has changed, then I'll update the docs to conform to the change.

**Comment by michaelmcandrew** [comment]
Hey there,

The other pragmatic dimension to this discussion that I want to point out...

When someone makes an improvement to the documentation, they pretty much always do this on master. The improvements are typically not 4.7 specific. They apply to 4.7, 4.6 and most likely a few versions before. But they are not back ported.

Back porting documentation improvements is not trivial but it is certainly possible, however no one is doing it.

So as it currently stands, we have the situation where it might be better to point someone on 4.6 to the 4.7/master documentation.

To be honest, there isn't a huge amount of difference today and my guess is that by the time there is a huge difference, people will be upgrading from 4.6 to 4.7 anyway (puts away his crystal ball)

Bottom line: 

* If 4.6 documentation is important to you, you might want to think more / think about resourcing back-porting of documentation improvements.
* If you don't have the resources for that, then its worth bearing in mind that in some areas, 4.7/stable is better for 4.6 users than 4.6.

**Comment by nganivet** [comment]
@michaelmcandrew I am no longer maintaining 4.6 as this is now in the hands of Square & Compucorp. I will point them to this thread so they can decide. We still have the nomenclature issue on the CiviCRM website and specifically on the download page. I will get to Josh on the topic since neither @totten nor @colemanw reacted on this.

**Comment by nganivet** [comment]
@seanmadsen - we have a decision from the core team on this topic:

 4.7: "Current" recommended version with the latest fixes and APIs ("Latest" would also be acceptable)
 4.6: "Long-Term Support" version with the least changes

Can you please implement these changes for the documentation - it seems like it would be multi-faceted with the build script, config files and documentation page at https://civicrm.org/documentation. I can help with these if you point me in the right direction.

Thanks.

**Comment by seanmadsen** [comment]
@nganivet yes, I will implement these changes, and I create #51 for this purpose. It might take me a couple weeks though because I'll need to do #50 first.

**Comment by nganivet** [comment]
Sure, thank you for that.
